tests/unit/test_blob.py

```diff
@@ -1906,10 +1911,14 @@ def _do_multipart_success(
 
         upload_url += "?" + urlencode(qs_params)
 
+        blob_data = b'{"name": "blob-name"}\r\n'
+        if metadata:
+            blob_data = b'{"name": "blob-name", "metadata": {"test": "test"}}\r\n'
+            self.assertEqual(blob._changes, set(["metadata"]))
         payload = (
             b"--==0==\r\n"
             + b"content-type: application/json; charset=UTF-8\r\n\r\n"
-            + b'{"name": "blob-name"}\r\n'
+            + blob_data
             + b"--==0==\r\n"
             + b"content-type: application/xml\r\n\r\n"
             + data_read
```

Comment:
This test needs to assert that `metadata` has been pushed to the blob's changes, too.

Reply:
I think assert for blob._changes is already there.
https://github.com/googleapis/python-storage/blob/cfefaaa3ef2b5e66bb9b60e9917095de39e40bce/tests/unit/test_blob.py#L1917